### PR TITLE
docs: Update release process

### LINKED ISSUE #2465: Make RELEASING.md a set of steps that can be followed
## Description

I'm finding [RELEASING.md](https://github.com/geeksforsocialchange/PlaceCal/blob/main/RELEASING.md) hard to follow. What's a thing I have to do, what's for my info, what bits matter and what don't? What's optional and what can't I do?

I think this needs rewriting to be a simple set of steps using the ["how to"](https://diataxis.fr/how-to-guides/) template.

## Motivation

RELEASING is currently imprecise about how to do a release and should be unambigous.

### PR DESCRIPTION
**Pull request by kimadactyl**
Simplify release process and make all steps explicit. Writing it out as I'm doing it, no hurry to merge.

Fixes #2465

### CODE REVIEW

**Comment by kimadactyl** [comment]
A bit unclear where to put screenshots/videos for new features, given they often close multiple tickets. Did my best to document how this is being done at the moment but happy to change it if you know a better way.

File: RELEASING.md [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
+Every pull request gets added to the latest draft release automatically. Validate them on the staging server before continuing.
 
-Go through each one and test any related tickets are truly fixed on the staging site and the new behaviour works as described. If there are any issues, make a call as to whether they need resolving before the release can happen or not. If they're not blocking issues, reopen the tickets or open new tickets for them instead and continue with the release.
+1. Validate that each pull request in the draft release does what it's meant to:
+   1. If it's a bug fix, check the bug linked in the PR is fixed on the [staging server](https://placecal-staging.org/). If it is, add a screenshot in a comment on the bug's issue ticket so the original reporter gets notified a fix is on the way.
+   2. If it's a new feature, make a screenshot or video of the feature working on the staging server and add it as a comment in the pull release.
+2. In both cases, open new issues as needed if these don't resolve the issue.
+3. Add the `verified` tag to the issues affected if you're happy the fix is working
```
Comment (katjam) [inline RELEASING.md:11]:
```suggestion
3. Add the `verified` label to resolved issues if you're happy the fix is working
```

**Review by katjam — APPROVED** [review]
Much improved. One tiny suggestion.
